packages/yaml/src/errors/index.ts

```diff
@@ -0,0 +1,65 @@
+import { interpolate } from '../texts';
+import type { StringMap } from '../texts';
+
+export interface YamlError {
+    code: string;
+    messageTemplate: string;
+}
+
+export const errors: Readonly<{ [err: string]: YamlError }> = Object.freeze({
+    yamlParsing: { code: 'yamlParsing', messageTemplate: 'Error parsing YAML document' },
+    nodeNotFound: { code: 'nodeNotFound', messageTemplate: 'Node not found at path: [{{- path }}]' },
+    nodeNotFoundMatching: {
+        code: 'nodeNotFoundMatching',
+        messageTemplate: 'Node not found at path: [{{- path }}, matching [{{- key }} = {{- value }}]'
+    },
+    nodeNotAMap: {
+        code: 'nodeNotAMap',
+        messageTemplate: 'Node path: [{{- path }}, matching [{{- key }} = {{- value }}], is not a map'
+    },
+    propertyNotFound: { code: 'propertyNotFound', messageTemplate: 'Node not found at path: [{{- path }}]' },
+    pathCannotBeEmpty: { code: 'pathCannotBeEmpty', messageTemplate: 'Path cannot be empty' },
+    pathDoesNotExist: { code: 'pathDoesNotExist', messageTemplate: 'Parent node does not exist at: [{{- path }}]' },
+    seqDoesNotExist: { code: 'seqDoesNotExist', messageTemplate: 'Sequence does not exist at: [{{- path }}]' },
+    tryingToAppendToNonSequence: {
+        code: 'tryingToAppendToNonSequence',
+        messageTemplate: 'Cannot append to non-sequence at: [{{- path }}]'
+    },
+    startNodeMustBeCollection: {
+        code: 'startNodeMustBeCollection',
+        messageTemplate: 'Starting node must be an array or an object'
+    },
+    scalarValuesDoNotHaveProperties: {
+        code: 'scalarValuesDoNotHaveProperties',
+        messageTemplate: 'Scalar values do not have properties'
+    }
+});
+
+/**
+ * Create a new error with `code` correctly set.
+ *
+ * @param options - Options object
+ * @param {YamlError} options.error - YAML error
+ * @param {string} options.message  - optional message for Error.message
+ * @param {StringMap} options.templateReplacements -  values for message template interpolation
+ * @returns {Error} - Error with correct code set
+ */
+export function newError({
```

Comment:
@ullasholla, if we need to work with different error types, would it make sense to create an own set of [custom errors](https://javascript.info/custom-errors) extending from `Error` and throw them accordingly?

Reply:
I'm undecided about that. Subclassing vs error codes were the clear options here. I picked error codes in the end because it gives me finer granularity, I didn't want to have to create multiple error classes to deal with all the different failure modes. In Nodejs, there seems to be a trend towards error codes as well: https://nodejs.org/dist/latest/docs/api/errors.html#errorcode

Reply:
@Klaus-Keller, take another look? I'm used both approaches - a custom error and error codes (for more granular handling)

Reply:
Link to updated code: https://github.com/SAP/open-ux-tools/blob/f34071a5d3752a6d6b7b65621c2f1c7ff6337381/packages/yaml/src/errors/index.ts#L44-L58

Usage: https://github.com/SAP/open-ux-tools/blob/f34071a5d3752a6d6b7b65621c2f1c7ff6337381/packages/odata-service-writer/src/index.ts#L88-L92

Reply:
Not a big thing, but why do you need the `newError` function? Cant you do the same in the constructor of `YAMLError`?

Reply:
> Not a big thing, but why do you need the `newError` function? Cant you do the same in the constructor of `YAMLError`?

I guess I could do that. I needed that `newError()` wrapper before as I had to create the error, set the code and then throw it. I'll take care of it

Reply:
Fixed in commit: https://github.com/SAP/open-ux-tools/pull/718/commits/a2b3f0661c8550dabf936bf0cb50be209527e2f3

Reply:
Thanks, I would have moved the execution of `interpolate` into the constructor, but this is really just me saving a few lines of code